esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -695,6 +695,31 @@ bool CWsESDLConfigEx::onPublishESDLBinding(IEspContext &context, IEspPublishESDL
     return true;
 }
 
+void setContainerizedEspBindingNameByPort(StringBuffer& espPort, StringBuffer& espBindingName)
+{
+    VStringBuffer xpath("services[@class='esp'][@port='%s']", espPort.str());
+    IPropertyTree* process = getGlobalConfigSP()->queryPropTree(xpath);
```

Comment:
You need to use

`  Owned<IPropertyTree> process = getGlobalConfigSP()->getPropTree(xpath);
`
The global config is valid for the duration of the assignment statement, but after that it may be deleted - and the pointer returned by queryPropTree() may be invalidated.  Alternatively you could assign getGlobalConfigSP() to a temporary as you had originally.